.github/workflows/ansible-release.yml

```diff
@@ -0,0 +1,150 @@
+name: Release Ansible package
+on:
+  workflow_dispatch:
+    inputs:
+      ansible-version:
+        description: >-
+          Release Version. Example: 11.1.0
+        required: true
+      ansible-major-version:
+        description: >-
+          Major Release Version. Example: 11
+        required: true
+env:
+  CI_COMMIT_MESSAGE: >-
+    Ansible ${{ inputs.ansible-version }}:
+    Dependencies, changelog and porting guide
+  ANSIBLE_VERSION: ${{ inputs.ansible-version }}
+
+jobs:
+  build:
+    name: Build Ansible (${{ inputs.ansible-version }})
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+      contents: write
+
+    steps:
+      - name: Check out antsibull
+        uses: actions/checkout@v3
+        with:
+          repository: ansible-community/antsibull
+          ref: main
+          path: antsibull
+
+      - name: Pre-create build directory
+        run: mkdir -p antsibull/build
+
+      # This is where the antsibull build-release role expects it by default
+      - name: Check out ansible-build-data under antsibull build directory
+        uses: actions/checkout@v3
+        with:
+          path: antsibull/build/ansible-build-data
+
+      - name: Set up Python 3.11
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
+
+      - name: Install dependencies
+        working-directory: antsibull
+        run: |
+          python3 -m pip install ansible-core
+          python3 -m pip install antsibull
+          ansible-galaxy install -r requirements.yml
+
+      - name: Checking out to a new branch
+        working-directory: antsibull/build/ansible-build-data
+        run: |
+          git checkout -b "publish-${ANSIBLE_VERSION}"
+
+      - name: Setting the user details
+        run: |
+          git config --global user.name "Github Actions"
+          git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+
+      # Run the playbook according to the current release process
+
+      - name: Building a release with the defaults
+        working-directory: antsibull
+        run: >-
+          ansible-playbook -vv playbooks/build-single-release.yaml
```

Comment:
@gotmax23 so I have a comment for you which I believe might need to be implemented in `antsibull`. This playbook builds the dists but doesn't perform the metadata validation. That must happen early in the process. Specifically, calling `twine check --strict` must succeed. My `pypi-upload` action is invoked late in the process and it does perform a non-strict check, which may fail the upload last moment which would be unpleasant. The PyPI backend might also reject uploads because of the problems with dists, some of which (but not all) can be caught in the strict mode.
Running the check as early as possible (on merges to `main` and pull requests) reveals metadata problems in a way that they can be fixed before attempting to build though this workflow.

So this workflow is the last line of defense that may reveal problems before the PR is created. I don't know if there's currently other places where it would be possible to inject this check, but perhaps putting it into that playbook would be useful. Something that could abort a release with broken meta.

Reply:
Yeah, adding it to the playbook (or _maybe_ shelling out to it from the python code) definitely makes sense to me. Can you open an issue in ansible's issue tracker?

Reply:
Done: #577. Marking this thread as resolved.